sdk/offchain/src/identifier.rs

```diff
@@ -51,6 +51,26 @@ impl Intent {
             self.amount,
         ))
     }
+
+    pub fn account_address(&self) -> &AccountAddress {
+        &self.account_address
+    }
+
+    pub fn subaddress(&self) -> &Subaddress {
+        &self.subaddress
+    }
+
+    pub fn currency_code(&self) -> &Option<String> {
+        &self.currency_code
+    }
+
+    pub fn amount(&self) -> &Option<u64> {
+        &self.amount
+    }
```

Comment:
Its not idiomatic to return an `&Option<T>` instead prefer returning `Option<&T>` for things that aren't Copy.

```
pub fn currency_code(&self) -> Option<&str> {
        self.currency_code.as_deref()
    }

    pub fn amount(&self) -> Option<u64> {
        self.amount // might need a `.copied()` call?
    }
```